polkadot-parachains/parachains-common/src/lib.rs

```diff
@@ -109,3 +116,56 @@ pub mod opaque {
 	/// Opaque block identifier type.
 	pub type BlockId = generic::BlockId<Block>;
 }
+
+//TODO: move DenyThenTry to polkadot's xcm module.
+/// Deny executing the xcm message if it matches any of the Deny filter regardless of anything else.
+/// If it passes the Deny, and matches one of the Allow cases then it is let through.
+pub struct DenyThenTry<Deny, Allow>
+where
+	Deny: ShouldExecute,
+	Allow: ShouldExecute,
+{
+	_deny: PhantomData<Deny>,
+	_allow: PhantomData<Allow>,
+}
+
+impl<Deny, Allow> ShouldExecute for DenyThenTry<Deny, Allow>
+where
+	Deny: ShouldExecute,
+	Allow: ShouldExecute,
+{
+	fn should_execute<Call>(
+		origin: &MultiLocation,
+		message: &mut Xcm<Call>,
+		max_weight: Weight,
+		weight_credit: &mut Weight,
+	) -> Result<(), ()> {
+		if Deny::should_execute(origin, message, max_weight, weight_credit).is_ok() {
+			return Err(())
+		}
+		Allow::should_execute(origin, message, max_weight, weight_credit)
+	}
+}
+
+// See issue #5233
+pub struct IsReserveTransferToRelayChain;
+impl ShouldExecute for IsReserveTransferToRelayChain {
+	fn should_execute<Call>(
+		_origin: &MultiLocation,
+		message: &mut Xcm<Call>,
+		_max_weight: Weight,
+		_weight_credit: &mut Weight,
+	) -> Result<(), ()> {
+		if message.0.iter().any(|inst| {
+			matches!(
+				inst,
+				DepositReserveAsset { dest: MultiLocation { parents: 1, .. }, .. } |
+					TransferReserveAsset { dest: MultiLocation { parents: 1, .. }, .. }
```

Comment:
You need to also specify the junction as well, otherwise you'd block other parachain destinations as well.

Reply:
Also, this seems to be denying XCMs that is trying to _send_ these two instructions to the relay chain, but it doesn't seem to restrict _executing_ these two kinds of instruction locally and then sending a `ReserveAssetDeposited` instruction onwards to the relay. I think the match condition here would thus need to be expanded to include that case.

Reply:
Should I be looking for a Transact with a ReserveAssetDeposited somewhere inside the encoded call? or do you mean a InitiateReserveWithdraw with a ReserveAssetDeposited inside it's xcm field?

Reply:
Have updated, does that look like what you were expecting?